### PR TITLE
Kahina/non secret array

### PR DESCRIPTION
**Pull request by kKahina**

### CODE REVIEW

File: src/boilerplate/orchestration/javascript/raw/toOrchestration.ts [hunk 1]
```diff
@@ -161,11 +161,15 @@ export const generateProofBoilerplate = (node: any) => {
         : [];  
     // we add any extra params the circuit needs
     node.parameters
+    
       .filter(
         (para: string) =>
           !privateStateNames.includes(para) && (
           !output.join().includes(`${para}.integer`) && !output.join().includes('msgValue')),
       )
+
```
Comment (SwatiEY) [inline src/boilerplate/orchestration/javascript/raw/toOrchestration.ts:170]: can you please remove these blank spaces

File: src/transformers/visitors/checks/interactsWithSecretVisitor.ts [hunk 2]
```diff
@@ -50,6 +74,14 @@ export default {
...
   },
 
+  FunctionDefinition: {
+    exit(path: NodePath) {
+      path.traversePathsFast(interactwithPublicSecret, {
+        publicPath: path,
+      });
+    },
+  },
+
```
Comment (SwatiEY) [inline src/transformers/visitors/checks/interactsWithSecretVisitor.ts:84]: Please check, we don't have a FunctionDefinition Node.

File: src/transformers/visitors/toOrchestrationVisitor.ts [hunk 3]
```diff
@@ -115,9 +115,22 @@ const findStatementId = (statements: any, ID: number) => {
 // i.e. public 'accessed' variables
 const addPublicInput = (path: NodePath, state: any, IDnode: any) => {
   const { node } = path;
+
   let { name } = path.scope.getReferencedIndicator(node, true) || path.node;
+//   if (name.includes('[') && name.includes(']')) {
```
Comment (SwatiEY) [inline src/transformers/visitors/toOrchestrationVisitor.ts:120]: Do we need this commented code?

File: src/transformers/visitors/toOrchestrationVisitor.ts [hunk 4]
```diff
@@ -299,22 +325,31 @@ const addPublicInput = (path: NodePath, state: any, IDnode: any) => {
...
+    if(IDnode) {
+      if (num_modifiers != 0)  {
+        if (IDnode.name === node.name){
+         IDnode.name += `_${num_modifiers}`;
+       } else {
+         IDnode.name =  `${node.name}_${num_modifiers}`;
+       }
+     }
     }
+
```
Comment (SwatiEY) [inline src/transformers/visitors/toOrchestrationVisitor.ts:345]: Please remove blank spaces

File: src/traverse/Scope.ts [hunk 5]
```diff
@@ -407,39 +409,42 @@ export class Scope {
...
-        : NodePath.getPath(referencingNode).getAncestorOfType('MemberAccess')
+        : NodePath.getPath(referencingNode)?.getAncestorOfType('MemberAccess')
             .node;
       return mappingKeyIndicatorOnly
         ? indicator.structProperties[memberAccessNode.memberName]
         : indicator;
     }
+   
 
+    //if (path.isConstantArray(referencingNode) && referencingNode.nodeType === 'variableDeclaration') return null;
```
Comment (SwatiEY) [inline src/traverse/Scope.ts:433]: Please remove this commented code if not required

File: src/transformers/visitors/checks/interactsWithSecretVisitor.ts [hunk 6]
```diff
@@ -34,6 +34,30 @@ const markSubtreeInteractsWithPublic = (thisPath: any, thisState: any) => {
...
+  const { node, scope } = thisPath;
+  if (!['Identifier', 'VariableDeclarationStatement'].includes(node.nodeType))
+    return;
+  const indicator = scope.getReferencedIndicator(node, true);
+  // we don't want to add itself as an interacted with path
+  if (indicator && thisState.secretPath.node.id !== node.id)
+    indicator.addSecretInteractingPath(thisState.secretPath);
+};
+
+const interactwithPublicSecret = (thisPath: any, thisState: any) => {
```
Comment (SwatiEY) [inline src/transformers/visitors/checks/interactsWithSecretVisitor.ts:47]: This function is not used anywhere, do we need this?

File: src/codeGenerators/orchestration/nodejs/toOrchestration.ts [hunk 7]
```diff
@@ -166,18 +170,24 @@ export default function codeGenerator(node: any, options: any = {}): any {
      return " ";
 
     case 'Assignment':
-      // To ensure the left hand side is always a general number, we generalise it here (excluding the initialisation in a for loop). 
+      // To ensure the left hand side is always a general number, we generalise it here (excluding the initialisation in a for loop).    
+    
       if (!node.isInitializationAssignment && node.rightHandSide.subType !== 'generalNumber'){
         if (['+=', '-=', '*='].includes(node.operator)) {
+
```
Comment (lydiagarms) [inline src/codeGenerators/orchestration/nodejs/toOrchestration.ts:177]: There's also a few blank lines here.

File: src/transformers/visitors/checks/interactsWithSecretVisitor.ts (hunk 6 above)
Comment (lydiagarms) [reply src/transformers/visitors/checks/interactsWithSecretVisitor.ts:47]: This is used on line 79.

File: test/contracts/Boolean-public.zol
Comment (lydiagarms) [inline test/contracts/Boolean-public.zol:1]: Why is this contract overwritten?

File: src/transformers/visitors/checks/interactsWithSecretVisitor.ts (hunk 6 above)
Comment (SwatiEY) [reply src/transformers/visitors/checks/interactsWithSecretVisitor.ts:47]: But are we using interactwithPublicSecret indicator anywhere ?

Comment (lydiagarms) [reply src/transformers/visitors/checks/interactsWithSecretVisitor.ts:47]: No, we use this function to set interactsWithSecret not interactsWithPublicSecret. We could change the name of this? Maybe to something like inferInteractsWithSecret?

File: src/transformers/visitors/checks/interactsWithSecretVisitor.ts [hunk 8]
```diff
@@ -50,6 +74,14 @@ export default {
     },
   },
 
+  FunctionDefinition: {
+    exit(path: NodePath) {
+      path.traversePathsFast(interactwithInferSecret, {
```
Comment (lydiagarms) [inline src/transformers/visitors/checks/interactsWithSecretVisitor.ts:79]: I think this should be inferInteractsWithSecret, as we are inferring whether an identifier interacts with secret.

**Review by lydiagarms — APPROVED** [review]

**Pull request merged**